CMakeLists.txt

```diff
@@ -139,32 +122,59 @@ endif( )
 set( ARMOR_LEVEL "${DEFAULT_ARMOR_LEVEL}" CACHE STRING "Enables increasingly expensive runtime correctness checks" )
 include( armor-config )
 
-# Find CUDA if the user wants a CUDA version.
-if (USE_CUDA)
-    find_package( CUDA REQUIRED )
-endif()
+# Deprecated USE_CUDA option
+if( DEFINED USE_CUDA )
+  if( USE_CUDA )
+    set( ENV{HIP_PLATFORM} nvidia )
+    message( DEPRECATED "USE_CUDA is deprecated (use environment variable HIP_PLATFORM=nvidia)" )
+  else( )
+    set( ENV{HIP_PLATFORM} amd )
+    message( DEPRECATION "USE_CUDA is deprecated (use environment variable HIP_PLATFORM=amd)" )
+  endif( )
+endif( )
 
 # Hip headers required of all clients; clients use hip to allocate device memory
-if( USE_CUDA)
+find_package( hip CONFIG PATHS ${ROCM_PATH} /opt/rocm )
+
+# support for cuda backend with hip < 6.0
+if( NOT hip_FOUND )
```

Comment:
The comment should really be hip < 5.7 because that's when the hip-nvidia-config.cmake implementation was added.

I'm not sure it's worth including this logic, since it's not going to work in a lot of situations (e.g., if you have both the AMD and NVIDIA tooling installed with ROCm 5.6, then the `find_package(hip)` will find the AMD platform tooling and you won't be able to build for the CUDA backend). If it was cost-free, then maybe that would be worth it anyway, but this fallback makes it more complicated to reason about the rest of the build.

If this is landing in ROCm 6.3, I don't think it's too unreasonable to require ROCm 5.7 to build for the CUDA backend at that point.